torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -386,13 +386,15 @@ class ShapePropagator {
         return;
       }
       case prim::ImplicitTensorToNum:
-      case prim::TensorToNum:
+      case prim::Bool:
+      case prim::Int:
+      case prim::Float:
         return; // correct num type is already set
       case prim::NumToTensor: {
-        if (node->input()->type()->isSubtypeOf(IntType::get())) {
+        TypePtr typ = node->input()->type();
+        if (typ->isSubtypeOf(IntType::get()) | typ->isSubtypeOf(BoolType::get())) {
           node->output()->setType(TensorType::create(at::kLong, at::kCPU, 0));
```

Comment:
I don't know when bool will get converted to Tensor, but i am surprised the tensor type is not kByte. If you create a tensor with: torch.tensor(True) the resulting type is ByteTensor.

Reply:
I am copying what was done in BoolToTensor below, which does turn it into a kLong. This behavior is likely not desired, but fixing it is a separate issue from this patch, which is suppose to be a refactor and not a behavior change.